### PR TITLE
feat: add CRP dataset builder with tiered quality selection

### PR DESCRIPTION
**Pull request by clarkemoyer**
## Summary

- Adds `scripts/analysis/create_crp_dataset.py` (1,026 lines) — builds the final N=200 CRP public dataset from Prolific Accepted responses using three-tier quality-based selection, then applies NIST 5-step de-identification for ScholarSphere deposit
- **Tier 1**: All Conservative Clean (APPROVED + every quality gate passed)
- **Tier 2**: All Flexible Clean surplus (APPROVED + all 3 IRIs + duration >= 480s)
- **Tier 3**: Quality-ranked fill from remaining Accepted using 6-indicator composite score (0–100 pts): IRI count (35), duration (20), reCAPTCHA (15), Prolific auth (15), full straightlining (8), partial straightlining (7)
- Generates detailed selection manifest with FLAG-SPEED breakdown showing which "Speed Only" responses were selected vs excluded
- Produces 6 output files: public CSV, selection manifest, ResponseId mapping, Prolific linkage, PII report, de-identification audit log

## Design Decisions

- Auth checks (Auth_LLM, Auth_Bots) weighted equally with reCAPTCHA and straightlining at 15 pts each — all three are independent platform-level authenticity signals
- FLAG-SPEED responses (all 3 IRIs correct, dur < 300s) are eligible for Tier 3 selection and explicitly reported in the manifest
- AUTO-EXCLUDE responses (IRI fail >= 2) rank lowest and are very unlikely to make the N=200 cut
- De-identification reuses the same NIST protocol as `deidentify_tabs_data.py` but operates only on the selected N=200 subset

## Test plan

- [ ] Verify Python syntax: `python3 -c "import ast; ast.parse(open('scripts/analysis/create_crp_dataset.py').read())"`
- [ ] Run with `--dry-run` flag against enriched Qualtrics CSV to validate selection logic without writing de-identified output
- [ ] Verify selection manifest shows correct tier counts and FLAG-SPEED detail
- [ ] Confirm N=200 with expected tier breakdown (~78 + ~45 + ~77)
- [ ] Verify public CSV has no PII columns (PROLIFIC_PID, IPAddress, etc.)
- [ ] Verify all 6 output files are generated

https://claude.ai/code/session_01VPDhLAKE66gUSpvFhBcRC1

### CODE REVIEW

File: scripts/analysis/create_crp_dataset.py [hunk 1]
```diff
@@ -0,0 +1,1026 @@
...
+def within_person_sd(responses: list[str]) -> float:
+    """Compute within-person SD for categorical responses."""
+    non_empty = [r for r in responses if r != ""]
+    if len(non_empty) < 2:
+        return float("nan")
+    unique_vals = list(dict.fromkeys(non_empty))
+    numeric = [unique_vals.index(r) for r in non_empty]
+    mean = sum(numeric) / len(numeric)
+    variance = sum((val - mean) ** 2 for val in numeric) / len(numeric)
+    return math.sqrt(variance)
```
[the remark this answers is not in the document]
Comment (clarkemoyer) [reply scripts/analysis/create_crp_dataset.py]: This implementation intentionally matches `tabs_v2_analysis.py:within_person_sd()` exactly — both use the index-of-first-appearance approach with a 0.5 threshold. Changing one without the other would create an inconsistency between the disposition pipeline and the CRP dataset builder (a response could be classified differently depending on which script runs).

The index-based SD is sufficient for its purpose: detecting near-zero-variance response patterns (straightlining/satisficing). The threshold was calibrated for this calculation across the existing pipeline.

No changes made.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/analysis/create_crp_dataset.py [hunk 2]
```diff
@@ -0,0 +1,1026 @@
...
+        "quality_score": quality,
+        "tier": None,       # assigned below
+        "selected": False,  # assigned below
+    }
+
+
+def select_crp_sample(
+    profiles: list[dict], target_n: int,
+) -> list[dict]:
+    """Apply tiered selection and return all profiles with tier/selected flags set."""
```
[the remark this answers is not in the document]
Comment (clarkemoyer) [reply scripts/analysis/create_crp_dataset.py]: Agreed — will add a focused test suite as a follow-up after validating the script against real enriched data. The tier assignment, FLAG-SPEED handling, and N=200 boundary conditions are the right targets for synthetic fixture tests.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/analysis/create_crp_dataset.py [hunk 3]
```diff
@@ -0,0 +1,1057 @@
...
+    print("  CRP DATASET BUILDER")
+    print(f"  Input:    {args.input_csv}")
+    print(f"  Target N: {args.target_n}")
+    print(f"  Output:   {args.output_dir}")
+    print(f"  Dry run:  {args.dry_run}")
+    print("=" * 78)
+
+    print("\n--- Loading enriched Qualtrics CSV ---")
+    headers, data = load_qualtrics_csv(str(args.input_csv))
+    idx = make_idx(headers)
```
[the remark this answers is not in the document]
Comment (clarkemoyer) [reply scripts/analysis/create_crp_dataset.py]: Fixed in fd4ba7d — added required-column check for PROLIFIC_PID, Prolific_Status, ResponseId, StartDate with exit code 2 and a message pointing to enrich_qualtrics_csv.py.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/analysis/create_crp_dataset.py [hunk 4]
```diff
@@ -0,0 +1,1057 @@
...
+    "Moderate Readiness/Capability": 3, "High Readiness/Capability": 4,
+    "Very High Readiness/Capability": 5,
+}
+MATURITY_SCALE = {
+    "Level 1: Initial/Ad Hoc": 1, "Level 2: Developing/Repeatable": 2,
+    "Level 3: Defined/Standardized": 3,
+    "Level 4: Managed/Quantitatively Managed": 4,
+    "Level 5: Optimizing/Innovating": 5,
+}
+
```
[the remark this answers is not in the document]
Comment (clarkemoyer) [reply scripts/analysis/create_crp_dataset.py]: Fixed in fd4ba7d — removed all three unused scale constants.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/analysis/create_crp_dataset.py [hunk 5]
```diff
@@ -0,0 +1,1069 @@
...
+            f.write(f"  Full text: {text_preview}\n\n")
+
+
+def deidentify_selected(
+    selected_rows: list[list[str]], headers: list[str], output_dir: Path,
+    skip_review: bool = False,
+) -> int:
+    """Run the full 5-step NIST de-identification on selected rows.
+
+    Returns True on success, False if PII verification fails.
```
[the remark this answers is not in the document]
Comment (clarkemoyer) [reply scripts/analysis/create_crp_dataset.py]: Fixed in ffad463 — docstring now reads "Returns: 0 on success, 1 if PII verification/review fails."

---
_Generated by [Claude Code](https://claude.ai/code)_
